lib/src/main/java/at/bitfire/cert4android/CustomCertStore.kt

```diff
@@ -0,0 +1,181 @@
+/***************************************************************************************************
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ **************************************************************************************************/
+
+package at.bitfire.cert4android
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.Log
+import kotlinx.coroutines.TimeoutCancellationException
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import org.conscrypt.Conscrypt
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.security.KeyStore
+import java.security.Security
+import java.security.cert.CertificateException
+import java.security.cert.X509Certificate
+import java.util.logging.Level
+import javax.net.ssl.SSLContext
+
+class CustomCertStore private constructor(
+    private val context: Context,
+    private val userTimeout: Long = 60000L
+) {
+
+    companion object {
+
+        const val KEYSTORE_DIR = "KeyStore"
+        const val KEYSTORE_NAME = "KeyStore.bks"
+
+        @SuppressLint("StaticFieldLeak")
+        private var instance: CustomCertStore? = null
+
+        @Synchronized
+        fun getInstance(context: Context): CustomCertStore {
+            instance?.let {
+                return it
+            }
+
+            val newInstance = CustomCertStore(context.applicationContext)
+            instance = newInstance
+            return newInstance
+        }
+
+    }
+
+    /** system default TrustStore */
+    private val systemKeyStore by lazy { Conscrypt.getDefaultX509TrustManager() }
+
+    /** custom TrustStore */
+    private val userKeyStoreFile by lazy { File(context.getDir(KEYSTORE_DIR, Context.MODE_PRIVATE), KEYSTORE_NAME) }
+    private val userKeyStore = KeyStore.getInstance(KeyStore.getDefaultType())!!
+
+    /** in-memory store for untrusted certs */
+    private var untrustedCerts = HashSet<X509Certificate>()
+
+    init {
+        // initialize Conscrypt
+        Security.insertProviderAt(Conscrypt.newProvider(), 1)
+
+        val version = Conscrypt.version()
+        Log.i(Cert4Android.TAG, "Using Conscrypt/${version.major()}.${version.minor()}.${version.patch()} for TLS")
```

Comment:
Is there any reason to not use `Cert4Android.log` here?

Reply:
Agree, can you please change it